MapboxMobileEventsTests/MMEEventTests.mm

```diff
@@ -129,6 +135,103 @@
             exceptionEventWithAllInfo should_not be_nil;
         });
     });
+
+    context(@"eventWithAttributes:error:", ^{
+        it(@"should not init with invalid attributes", ^{
+            NSError *error = nil;
+            MMEEvent *invalid = [MMEEvent eventWithAttributes:@{@"Invalid": [NSError errorWithDomain:NSCocoaErrorDomain code:0 userInfo:nil]} error:&error];
+
+            invalid should be_nil;
+            error should_not be_nil;
+            error.code should equal(MMEErrorEventInit);
+        });
+
+        it(@"should init with nil attributes for initWithCoder:", ^{
+            NSError *error = nil;
+            MMEEvent *invalid = [MMEEvent eventWithAttributes:MMEEvent.nilAttributes error:&error];
+
+            invalid.attributes should be_nil;
+            error should be_nil;
+        });
+
+        it(@"should not init without MMEEventKeyEvent", ^{
+            NSError *error = nil;
+            MMEEvent *invalid = [MMEEvent eventWithAttributes:@{@"foo":@"bar"} error:&error];
+
+            invalid should be_nil;
+            error should_not be_nil;
+            error.code should equal(MMEErrorEventInit);
+        });
+
+        it(@"should contain an exceptional dictionary", ^{
+            NSError *error = nil;
+            MMEEvent *exceptional = [MMEEvent eventWithAttributes:[MMEExceptionalDictionary dictionaryWithDictionary:@{@"foo":@"bar"}] error:&error];
+
+            exceptional should be_nil;
+            error should_not be_nil;
+            error.code should equal(MMEErrorEventInit);
+        });
+    });
+
+    context(@"eventWithAttributes:", ^{
+        it(@"should return nil with invalid attributes:", ^{
+            MMEEvent *attributed = [MMEEvent eventWithAttributes:@{@"invalid":@"attributes"}];
+
+            attributed should be_nil;
+        });
+    });
+
+    context(@"carplayEvent", ^{
+        it(@"should create a carplay event", ^{
+            MMEEvent *carplay = [MMEEvent carplayEventWithName:MMEventTypeNavigationCarplayConnect attributes:testAttrs];
+
+            carplay should_not be_nil;
+        });
+    });
+
+    context(@"isEqualToEvent:", ^{
+
+        it(@"should not be true for two consective events with the same attributes, or for nil", ^{
+            NSDictionary *eventAttributes = @{MMEEventKeyEvent: @"test.event"};
+            MMEEvent *firstEvent = [MMEEvent eventWithAttributes:eventAttributes];
+            MMEEvent *secondEvent = [MMEEvent eventWithAttributes:eventAttributes];
```

Comment:
Could this also be moved into a `cbef`(`beforeEach`) block?

Reply:
broke this mono block up into a bunch of individual tests on the events